packages/debug/src/browser/debug-prefix-configuration.ts

```diff
@@ -107,18 +107,42 @@ export class DebugPrefixConfiguration implements CommandContribution, CommandHan
         });
     }
 
-    getPicks(filter: string, token: CancellationToken): QuickPicks {
+    async getPicks(filter: string, token: CancellationToken): Promise<QuickPicks> {
         const items: QuickPickItem[] = [];
-        const configurations = this.debugConfigurationManager.all;
-        Array.from(configurations).forEach(config => {
+        const configurationManager = this.debugConfigurationManager;
+        const configurations = configurationManager.all;
+
+        for (const config of configurations) {
             items.push({
                 label: config.configuration.name,
                 description: this.workspaceService.isMultiRootWorkspaceOpened
                     ? this.labelProvider.getName(new URI(config.workspaceFolderUri))
                     : '',
                 execute: () => this.runConfiguration(config)
             });
-        });
+        }
+
+        // Resolve dynamic configurations from providers
+        const configurationsByType = await configurationManager.provideDynamicDebugConfigurations();
+        for (const typeConfigurations of configurationsByType) {
+            const type = typeConfigurations.type;
+            const dynamicConfigurations = typeConfigurations.configurations;
+            if (dynamicConfigurations.length > 0) {
+                items.push({
+                    label: type,
```

Comment:
minor: (removing the variable)

```suggestion
                    label: typeConfigurations.type,
```